src/Microsoft.ML.TimeSeries/SrCnnEntireTransformer.cs

```diff
@@ -0,0 +1,903 @@
+                ShortName = "mode", SortOrder = 303)]
+            public SrCnnDetectMode SrCnnDetectMode = SrCnnDetectMode.AnomalyOnly;
+
+            [Argument(ArgumentType.AtMostOnce, HelpText = "The sensitivity of boundaries.",
+                ShortName = "sens", SortOrder = 404)]
+            public double Sensitivity = 99;
+        }
+
+        internal readonly string InputColumnName;
+
+        internal readonly string OutputColumnName;
+
+        internal double Threshold { get; }
+
+        internal int BatchSize { get; }
+
+        internal SrCnnDetectMode SrCnnDetectMode { get; }
+
+        internal double Sensitivity { get; }
+
+        internal int OutputLength { get; }
+
+        internal SrCnnEntireModeler AnomalyModel;
+
+        private static VersionInfo GetVersionInfo()
+        {
+            return new VersionInfo(
+                modelSignature: "SRENTRNS",
+                verWrittenCur: 0x00010001, // Initial
+                verReadableCur: 0x00010001,
+                verWeCanReadBack: 0x00010001,
+                loaderSignature: LoaderSignature,
+                loaderAssemblyName: typeof(SrCnnEntireTransformer).Assembly.FullName);
+        }
+
+        internal SrCnnEntireTransformer(IHostEnvironment env, Options options, IDataView input)
+            :base(Contracts.CheckRef(env, nameof(env)).Register(LoaderSignature), new[] { (options.Target, options.Source) })
+        {
+            InputColumnName = options.Source;
+            OutputColumnName = options.Target;
+
+            Host.CheckUserArg(options.Threshold >= 0 && options.Threshold <= 1, nameof(Options.Threshold), "Must be in [0,1].");
+            Threshold = options.Threshold;
+
+            Host.CheckUserArg(options.BatchSize == -1 || options.BatchSize >= 12, nameof(Options.BatchSize), "BatchSize must be -1 or no less than 12.");
+            BatchSize = options.BatchSize;
+
+            SrCnnDetectMode = options.SrCnnDetectMode;
+            if (SrCnnDetectMode.Equals(SrCnnDetectMode.AnomalyOnly))
+            {
+                Sensitivity = options.Sensitivity;
+                OutputLength = 3;
+            }
+            else if (SrCnnDetectMode.Equals(SrCnnDetectMode.AnomalyAndMargin))
+            {
+                Host.CheckUserArg(options.Sensitivity >= 0 && options.Sensitivity <= 100, nameof(Options.Sensitivity), "Must be in [0,100].");
+                Sensitivity = options.Sensitivity;
+                OutputLength = 7;
+            }
+            else if (SrCnnDetectMode.Equals(SrCnnDetectMode.AnomalyAndExpectedValue))
+            {
+                Sensitivity = options.Sensitivity;
+                OutputLength = 4;
+            }
+            else
+            {
+                OutputLength = 0;
+            }
+
+            AnomalyModel = new SrCnnEntireModeler(Host, this, input);
+        }
+
+        private SrCnnEntireTransformer(IHost host, ModelLoadContext ctx)
```

Comment:
>SrCnnEntireTransformer [](start = 16, length = 22)

Please add a comment inside this method and in the `SaveModel` method detailing the binary format (for an example of what I mean, see here: [https://github.com/dotnet/machinelearning/blob/master/src/Microsoft.ML.ImageAnalytics/ImageResizer.cs#L231-L238](https://github.com/dotnet/machinelearning/blob/master/src/Microsoft.ML.ImageAnalytics/ImageResizer.cs#L231-L238)).

Reply:
> > SrCnnEntireTransformer [](start = 16, length = 22)
> 
> Please add a comment inside this method and in the `SaveModel` method detailing the binary format (for an example of what I mean, see here: https://github.com/dotnet/machinelearning/blob/master/src/Microsoft.ML.ImageAnalytics/ImageResizer.cs#L231-L238).

Added.